ruby/README.md

```diff
@@ -16,58 +16,109 @@ Notice in `hello.rb`, we add the following so it uses the vendored gems:
 require_relative 'bundle/bundler/setup'
 ```
 
+### 2. Test locally
+
 Now test it locally:
 
 ```sh
 docker run --rm -it -e "PAYLOAD_FILE=hello.payload.json" -v "$PWD":/worker -w /worker iron/ruby ruby hello.rb
 ```
 
+The PAYLOAD_FILE environment variable is passed in to your worker automatically and tells you
+where the payload file is. Our [client libraries](http://dev.iron.io/worker/libraries/) help you load the special environment variables automatically.
+
 Boom, it works! And now that it works, we know it will work on IronWorker.
 
-Let's package it up:
+### 3. Package your code
+
+Let's package it up inside a Docker image and upload it to a Docker Registry. Copy the [Dockerfile](https://github.com/iron-io/dockerworker/blob/master/ruby/Dockerfile) from this repository
+and modify the ENTRYPOINT line to run your script, but for this example, it's all ready to go:
 
 ```sh
-zip -r hello.zip .
+docker build -t username/hello.rb:0.0.1 .
 ```
 
-Then upload it:
+That's just a standard `docker build` command. The 0.0.1 is the version which you can update
+whenever you make changes to your code.
+
+Test your image, just to be sure you created it correctly:
 
 ```sh
-iron worker upload --name hello --zip hello.zip iron/ruby ruby hello.rb
+docker run --rm -it -e "PAYLOAD_FILE=hello.payload.json" username/hello.rb:0.0.1
+```
+
+### 4. Push it to Docker Hub
+
+Push it to Docker Hub:
+
+```sh
+docker push username/hello.rb
 ```
 
-And finally queue up a job for it!
+### 4. Register your image with Iron
+
+Ok, we're ready to run this on Iron now, but first we have to let Iron know about the
+image you just pushed to Docker Hub:
 
 ```sh
-iron worker queue --payload-file hello.payload.json --wait hello
+iron worker register username/hello.rb:0.0.1
+```
+
+### 5. Queue / Schedule jobs for your image
+
+Now you can start queuing jobs or schedule recurring jobs for your image. Let's quickly
+queue up a job to try it out.
+
+```sh
+iron worker queue --payload-file hello.payload.json --wait username/hello.rb
 ```
 
 The `--wait` parameter waits for the job to finish, then prints the output.
 You will also see a link to [HUD](http://hud.iron.io) where you can see all the rest of the task details along with the log output.
 
-<!--
-## Bundling the worker inside a Docker image
+Read the API docs to see how to queue jobs from your code or how to schedule them:
+http://dev.iron.io/worker/reference/api/
 
-Follow the same steps above to run/test your worker. But instead of zipping it up, do the following:
+## Private images
 
-Package it up inside an image and send it off to Docker HUB, see the `Dockerfile` for reference:
+If you want to keep your code private and use a [private Docker repository](https://docs.docker.com/docker-hub/repos/#private-repositories), you just need
+to let Iron know it's private.
 
 ```sh
-docker build -t treeder/hello.rb:latest .
+iron docker login -e YOUR_DOCKER_HUB_EMAIL -p YOUR_DOCKER_HUB_PASSWORD
```

Comment:
should leave a note here about how we store these credentials on our machines and what guarantees we make about their fidelity